runelite-client/src/main/java/net/runelite/client/plugins/bosslogger/BossLoggerPlugin.java

```diff
@@ -0,0 +1,1365 @@
+	{
+		// Only update if our plugin config was changed
+		if (!event.getGroup().equals("bosslogger"))
+		{
+			return;
+		}
+
+		handleConfigChanged(event.getKey());
+	}
+
+	// Searches chat messages for kill count value or pet drop
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() != ChatMessageType.SERVER && event.getType() != ChatMessageType.FILTERED)
+		{
+			return;
+		}
+
+		String chatMessage = event.getMessage();
+
+		// Barrows KC
+		if (chatMessage.startsWith("Your Barrows chest count is"))
+		{
+			Matcher m = NUMBER_PATTERN.matcher(Text.removeTags(chatMessage));
+			if (m.find())
+			{
+				killcountMap.put("BARROWS", Integer.valueOf(m.group()));
+				return;
+			}
+		}
+
+		// Raids KC
+		if (chatMessage.startsWith("Your completed Chambers of Xeric count is:"))
+		{
+			Matcher m = NUMBER_PATTERN.matcher(Text.removeTags(chatMessage));
+			if (m.find())
+			{
+				killcountMap.put("RAIDS", Integer.valueOf(m.group()));
+				return;
+			}
+		}
+
+		// Raids KC
+		if (chatMessage.startsWith("Your completed Theatre of Blood count is:"))
+		{
+			Matcher m = NUMBER_PATTERN.matcher(Text.removeTags(chatMessage));
+			if (m.find())
+			{
+				killcountMap.put("RAIDS 2", Integer.valueOf(m.group()));
+				return;
+			}
+		}
+
+		// Handle all other boss
+		Matcher boss = BOSS_NAME_PATTERN.matcher(Text.removeTags(chatMessage));
+		if (boss.find())
+		{
+			String bossName = boss.group(1);
+			Matcher m = NUMBER_PATTERN.matcher(Text.removeTags(chatMessage));
+			if (!m.find())
+				return;
+			int KC = Integer.valueOf(m.group());
+			killcountMap.put(bossName.toUpperCase(), KC);
+		}
+
+		// Pet Drop
+		Matcher pet1 = PET_RECEIVED_PATTERN.matcher(Text.removeTags(chatMessage));
+		Matcher pet2 = PET_RECEIVED_INVENTORY_PATTERN.matcher(Text.removeTags(chatMessage));
+		if (pet1.find() || pet2.find())
+		{
+			gotPet = true;
+		}
+	}
+
+	private void resetStoredData()
+	{
+		for (Tab tab : Tab.values())
+		{
+			lootMap.put(tab.getBossName().toUpperCase(), new ArrayList<>());
+		}
+	}
+
+	private void updateTabData()
+	{
+		// Do nothing is panel isn't enabled
+		if (!bossLoggerConfig.showLootTotals())
+		{
+			return;
+		}
+
+		for (Tab tab : Tab.values())
+		{
+			// Only update tabs if the tabs are being shown.
+			if (isBeingRecorded(tab.getName()))
+			{
+				panel.updateTab(tab.getName());
+			}
+		}
+	}
+
+	private void init()
+	{
+		// Create maps for easy management of certain features
+		Map<String, Boolean> mapRecording = new HashMap<>();
+		Map<String, ArrayList<LootEntry>> mapLoot = new HashMap<>();
+		Map<String, Integer> mapKillcount = new HashMap<>();
+		Map<String, String> mapFilename = new HashMap<>();
+		for (Tab tab : Tab.values())
+		{
+			String bossName = tab.getBossName().toUpperCase();
+			// Is Boss being recorded?
+			mapRecording.put(bossName, isBeingRecorded(tab.getName()));
+			// Loot Entries by Tab Name
+			ArrayList<LootEntry> array = new ArrayList<LootEntry>();
```

Comment:
Nitpicking but declare the interface, provide an implementation

`List<LootEntry> array = new ArrayList<LootEntry>();`